tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,218 @@
+    record_info('Guest ID', $vm_id);
+    my $vm_ip = esxi_vm_public_ip($vm_id);
+    record_info('Guest IP', $vm_ip);
+
+    record_info('Power Manangement Tests');
+    power_management_tests($vm_name, $vm_id, $vm_ip);
+    record_info('Networking Tests');
+    networking_tests($vm_id, $vm_ip);
+    record_info('Clock Sync Tests');
+    clock_sync_tests($vm_name, $vm_id, $vm_ip);
+}
+
+sub get_timelc {
+    my ($time, $timelc, $sec, $min, $hour, $mday, $mon, $year);
+    $time = $_[0];
+
+    my @time = reverse(split /-|:|\s|\//, $time);
+    foreach (@time) {
+        s/^0//;
+        $timelc .= $_ . ",";
+    }
+    $timelc =~ s/,$//;
+    ($sec, $min, $hour, $mday, $mon, $year) = split /,/, $timelc;
+    $year -= 1900;
+    $mon  -= 1;
+    $timelc = timelocal($sec, $min, $hour, $mday, $mon, $year);
+
+    return $timelc;
+}
+
+sub power_management_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+
+    select_console('svirt');
+
+    # Guest Power Off
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Power Off');
+        esxi_vm_power_off($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Power On
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered off/) {
+        record_info('Guest Power On');
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Reset
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Reset');
+        esxi_vm_power_reset($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Restart
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Restart');
+        esxi_vm_power_reboot($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Shutdown
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Shutdown');
+        esxi_vm_power_shutdown($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+}
+
+sub networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered off/) {
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    reset_consoles;
```

Comment:
I am not sure whether I get this code block(L131~138) correctly. Would you please help explain what's the purpose for the block and what are done to achieve this? BTW is this the part that are blocked by the svirt backend issue which you mentioned in today's team meeting?

Reply:
The purpose of this code block is that running ping commands inside the vm to check its guest network. We have to select vm console and do some login actions after vm booted up, then we could run those ping commands.
Actually, the ping command(assert_script_run("ping -c3 www.suse.com")) was blocked after vm shutdown and booted up(Referring defect #97517).